### ISSUE TITLE
Connect default_room to speclist via random lookup function

### ISSUE THREAD

**Issue by CCalderon01** [uchicago-cs/chiventure#665]
Use the make_default_room() function to lookup and append "random" rooms and items to a speclist. Given a bucket string, load the respective roomspec hash with all the allowed rooms and items and select a combination of these rooms and items to be added to a input speclist. This will be used especially in the autogeneration module.

**Comment by CCalderon01** [comment]
Commit: fdb5412ae0d
I implemented and added descriptions for 5 functions (speclist_from_hash, random_room_content, random_room_lookup, random_item_lookup, random_items). These functions are intermediaries between the default_room and autogeneration module. In the default_room module, we generate a "master" roomspec hash table with all the acceptable items and rooms for a given bucket. Now, we need to return one room and n number of items loaded into the room -- we can do this by choosing a "random" combination of allowed rooms and their allowed items. See autogeneration.h for more details.

**Comment by CCalderon01** [comment]
Commit: 8dbabd0 on openworld/speclist branch
I refactored my code and got rid of the random_room_content function since it was very repetitive and I didn't need to make a new function just to call random_room_lookup. I wrote tests for all these functions and fixed all the NULL errors.

**Comment by CCalderon01** [comment]
I linked pull request #710 to this issue since I'm now ready to merge my changes into the dev branch. In summary, there are 4 functions I want to add to the autogenerate module. First, speclist_from_hash appends all the roomspecs in a given roomspec hash to a speclist. This is important because we generate rooms using a speclist, not a roomspec hash. Next, random_items iterates "n" number of times through a given item hash to copy 1 item from a source hash table to a destination hash table. This function is used in random_item_lookup. Random_item_lookup takes a speclist roomspec entry, specifically its item hash parameter, and calls on random_items a "random" number of times to add n total items to the roomspec's item hash. Putting this all together, in random_room_lookup, we "randomly" select a roomspec from a given speclist and load a new set of items into using random_item_lookup. Finally, these will be used in the generate functions to create a new room.

**Comment by CCalderon01** [comment]
After Bryan's pull request (https://github.com/uchicago-cs/chiventure/pull/681) was merged successfully, I modified my code to match the new logic that the generate functions have. I added in the unique room identifiers.

**Comment by CCalderon01** [comment]
Commit 107988d on openworld/speclist branch
I removed the unique identifiers by refactoring the code and adding parameters to the roomspec_to_room and room_generate functions. My original 4 functions (speclist_from_hash, random_room_lookup, random_item_lookup, random_items) were minimally changed, but roomspec_to_room, room_generate, multi_room_generate were changed in terms of their input/scope. The logic in all functions remained the same.
